theme/views.py

```diff
@@ -291,6 +293,36 @@ def send_verification_mail_for_email_update(request, user, new_email, verificati
                        context=context)
 
 
+def login(request, template="accounts/account_login.html",
+          form_class=LoginForm, extra_context=None):
+    """
+    Login form - customized from Mezzanine login form so that quota warning message can be
+    displayed when the user is logged in.
+    """
+    form = form_class(request.POST or None)
+    if request.method == "POST" and form.is_valid():
+        login_msg = "Successfully logged in"
+        authenticated_user = form.save()
+        qmsg = QuotaMessage.objects.first()
+        soft_limit = qmsg.soft_limit_percent
+        for uq in authenticated_user.quotas.all():
+            percent = uq.used_value*100/uq.allocated_value
+            if percent >= soft_limit:
+                qmsg = QuotaMessage.objects.first()
+                msg_template_str = ' - {}{}'.format(qmsg.warning_content_prepend, qmsg.content)
+                login_msg += msg_template_str.format(used=uq.used_value,
+                                                     unit=uq.unit,
+                                                     allocated=uq.allocated_value,
+                                                     zone=uq.zone,
+                                                     percent=percent)
+        info(request, _(login_msg))
```

Comment:
@pkdash I agree. Will add the warning message on user profile page. I also need to add quota information on user profile page, so will do them together.